forte/helpers/helpers.h

```diff
@@ -182,10 +185,34 @@ void apply_permutation_in_place(std::vector<T>& vec, const std::vector<std::size
     }
 }
 
+/**
+ * @brief Apply in-place matrix transposition based on the algorithm of Catanzaro, Keller, Garland.
+ * @param data the matrix stored in row-major format
+ * @param m the number of rows of the matrix
+ * @param n the number of columns of the matrix
+ *
+ * See Algorithm 1 of DOI: 10.1145/2555243.2555253.
+ * Also see https://github.com/bryancatanzaro/inplace
+ */
+void matrix_transpose_in_place(double* data, const size_t m, const size_t n);
+
+void push_to_psi4_env_globals(double value, const std::string& label);
+
 namespace math {
 /// Return the number of combinations of n identical objects
 size_t combinations(size_t n, size_t k);
 
+/// Compute the greatest common divisor
+inline int gcd(int a, int b) {
```

Comment:
Any good reason for not using the std::gcd functions (https://en.cppreference.com/w/cpp/numeric/gcd)?

Reply:
Extended GCD is needed for the more optimized in-place matrix transpose algorithm. But currently, the numeric one should be good.